From 69d8694b9d9b9098b8434e7b4f219b1f01ce1445 Mon Sep 17 00:00:00 2001
From: Tim Hartley <timothy.d.hartley@intel.com>
Date: Wed, 31 Jul 2013 09:48:54 -0700
Subject: Dalvik: Fix for libdvm_assert build

BZ: 127516

A trivial patch to fix an assert error.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I42e6705197fe479c6a2547761b40cc8928212126
Orig-MCG-Change-Id: Icb66b3da16bfc6a148aaaa7df00997b330c8f130
Signed-off-by: Tim Hartley <timothy.d.hartley@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Reviewed-on: http://android.intel.com:8080/123670
Reviewed-by: Popov, Ivan G <ivan.g.popov@intel.com>
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/Dataflow.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/Dataflow.cpp b/vm/compiler/Dataflow.cpp
index 3fefe8f..3051280 100644
--- a/vm/compiler/Dataflow.cpp
+++ b/vm/compiler/Dataflow.cpp
@@ -2569,7 +2569,7 @@ bool dvmCompilerFindInductionVariablesHelper(CompilationUnit *cUnit,
                     }
 
                     /* Guaranteed to find an element */
-                    assert(i < ivList->numUsed);
+                    assert(i < ivList.numUsed);
 
                     ivInfo->ssaReg = mir->ssaRep->defs[0];
                     ivInfo->basicSSAReg = ivInfoOld->basicSSAReg;
-- 
1.7.4.1

